### PR TITLE
Doxygen updates for upds3.f, upftbv.f,  ups.f, and uptdd.f

### PR DESCRIPTION
**Pull request by jack-woollen**
Doxygen updates.    

Part of https://github.com/NOAA-EMC/NCEPLIBS-bufr/issues/246

### CODE REVIEW

File: src/uptdd.f [hunk 1]
```diff
@@ -1,62 +1,35 @@
 C> @file
-C> @author WOOLLEN @date 1994-01-06
-      
-C> THIS SUBROUTINE RETURNS THE BIT-WISE REPRESENTATION OF THE
-C>   FXY VALUE CORRESPONDING TO, SEQUENTIALLY, A PARTICULAR (IENT'th)
-C>   "CHILD" MNEMONIC OF A TABLE D SEQUENCE ("PARENT") MNEMONIC.
+C> @brief this subroutine returns the fxy value of an element in a sequence
```
Comment (edwardhartnett) [inline src/uptdd.f]: Capitalize and punctuation

File: src/uptdd.f [hunk 2]
```diff
@@ -1,62 +1,35 @@
...
-C>    THIS ROUTINE CALLS:        BORT     IUPM
-C>    THIS ROUTINE IS CALLED BY: NEMTBD   RESTD
-C>                               Normally not called by any application
-C>                               programs.
+C> @param[in] id - integer: positional index of parent mnemonic within internal bufr table d array tabd
+C> @param[in] lun - integer: i/o stream index into internal memory arrays
+C> @param[in] ient - integer: ordinal indicator of child mnemonic to return from within tabd(id,lun) sequence:
+C> @param[out] iret - integer: return value (see remarks)
 C>
+C> @author woollen @date 1994-01-06
```
Comment (edwardhartnett) [inline src/uptdd.f]: Capitalize your name.

File: src/ups.f [hunk 3]
```diff
@@ -1,34 +1,29 @@
 C> @file
-C> @author J @date 2012-03-02
-	
-C> THIS FUNCTION UNPACKS A REAL*8 USER VALUE FROM A PACKED
-C>   BUFR INTEGER BY APPLYING THE PROPER SCALE AND REFERENCE VALUES.
-C>   NORMALLY THE SCALE AND REFERENCE VALUES ARE OBTAINED FROM INDEX
-C>   NODE OF THE INTERNAL JUMP/LINK TABLE ARRAYS ISC(*) AND IRF(*);
-C>   HOWEVER, THE REFERENCE VALUE IN IRF(*) WILL BE OVERRIDDEN IF A
-C>   2-03 OPERATOR IS IN EFFECT FOR THIS NODE.
+C> $brief unpacks a real*8 user value from a packed bufr integer
```
Comment (edwardhartnett) [inline src/ups.f]: Capitalize and punctuate.

File: src/ups.f [hunk 4]
```diff
@@ -1,34 +1,29 @@
...
-C> USAGE:    UPS (IVAL,NODE)
-C>   INPUT ARGUMENT LIST:
-C>     IVAL      - INTEGER: PACKED BUFR INTEGER
-C>     NODE      - INTEGER: INDEX INTO INTERNAL JUMP/LINK TABLES
-C>
-C>   OUTPUT ARGUMENT LIST:
-C>     UPS       - REAL*8: USER VALUE
+C> @author J. Ator @date 2012-03-02
+
+C> this function unpacks a real*8 user value from a packed
```
Comment (edwardhartnett) [inline src/ups.f]: Capitalize and punctuate.

File: src/uptdd.f (hunk 1 above)
Comment (edwardhartnett) [inline src/uptdd.f]: Capitalize and punctuate.

**Comment by edwardhartnett** [comment]
Please use a more descriptive PR title, and also mention the issue in the description. In this case, since you are working on an issue, but not finishing it (i.e. closing it), you can say "Part of #246" in your description.

No need to list the files modified in the description, since they already appear in the review. Instead, describe the purpose of the PR.

Following these guidelines leaves a clear record of our work and provides hyper-linked documentation for our releases.

See the other PRs for examples.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: src/uptdd.f (hunk 1 above)
Comment (jack-woollen) [reply src/uptdd.f]: Got it.

File: src/upftbv.f [hunk 5]
```diff
@@ -1,39 +1,34 @@
...
-C>                     "On" (i.e. set to "1") in VAL
-C> @param[out] NIB  -- integer: Number of bit numbers returned in
-C>                     IBIT
-C>
 C> @remarks
 C> - This subroutine is the logical inverse of function pkftbv().
 C> - According to the WMO standard, bits within a bit field are
 C> numbered from left to right, so bit #1 is always the high-order
 C> (i.e. most significant) bit in any bit field.
+C> @author J. Ator @date 2005-11-29
```
Comment (edwardhartnett) [inline src/upftbv.f]: Author at bottom.

File: src/usrtpl.f [hunk 6]
```diff
@@ -1,53 +1,31 @@
...
+C> This subroutine stores the subset template into internal
+C> subset arrays in module usrint for cases of node expansion
+C> (i.e. when the node is either a table a mnemonic or a delayed
+C> replication factor).
 C>
+C> @param lun - integer: i/o stream index into internal memory arrays
+C> @param invn - integer: starting jump/link table index of the node to be expanded within the subset template
+C> @param nbmp - integer: number of times by which invn is to be expanded (i.e. number of replications of node)
+C>
+C> @author woollen @date 1994-01-06
```
Comment (edwardhartnett) [inline src/usrtpl.f]: Capitalize name.

File: src/wrdxtb.f [hunk 7]
```diff
@@ -1,36 +1,33 @@
...
-C> @author J. Ator
-C> @date 2009-03-23
-C>
-C> @param[in] LUNDX   -- integer: Fortran logical unit number
-C>                       associated with DX BUFR table information
-C>                       to be written out
-C> @param[in] LUNOT   -- integer: Fortran logical unit number of
-C>                       BUFR file to which messages are to be
-C>                       written
+C> @param[in] lundx - integer: Fortran logical unit number associated with DX BUFR table information to be written out
```
Comment (edwardhartnett) [inline src/wrdxtb.f]: I usually don't let the lines get this long.

**Comment by jack-woollen** [comment]
@edwardhartnett @jbathegit Another question: does the capitalize and punctuate also apply to the program log and in/out parameter text?

**Comment by edwardhartnett** [comment]
No for history logs since we are removing them.

Yes for parameters.

File: src/moda_ivttmp.F [hunk 8]
```diff
@@ -1,18 +1,22 @@
...
+C> @author Woollen @date 1994-01-06
 
-C> ??? ~description~
+C> These arrays provide working space in several subprograms (usrtpl and
+C> ufbcup) which manipulate the contents of the internal subset buffers
+C> where the contents of an evolving BUFR message are sccumulated and 
+C> stored under user control prior to being written out.
 C>
-C> @author ??? @date ???
+C>  TTMP - space used to store TAG elementS for new sections of a growing subset buffer
```
Comment (edwardhartnett) [inline src/moda_ivttmp.F]: The module variable need to be documented with the variables, not up here.

File: src/moda_ivttmp.F [hunk 9]
```diff
@@ -1,18 +1,22 @@
...
+C>
+C> @author Woollen @date 1994-01-06
 
 	MODULE MODA_IVTTMP
 
-C>  ??? description of TTMP
-	  CHARACTER*10, ALLOCATABLE :: TTMP(:)
-C>  ??? description of ITMP
-	  INTEGER, ALLOCATABLE :: ITMP(:) 
-C>  ??? description of VTMP
```
Comment (edwardhartnett) [inline src/moda_ivttmp.F:15]: This is where the module vars must be documented. Each one gets it's own doxygen block.

File: src/uptdd.f [hunk 10]
```diff
@@ -1,62 +1,35 @@
...
-C>              corresponding to the IENT'th child mnemonic of
-C>              TABD(ID,LUN)
-C>    END IF
+C> @author Woollen @date 1994-01-06
+
+C> This subroutine returns the bit-wise representation of the fxy value corresponding to, sequentially, 
+C> a particular (ient'th) "child" mnemonic of a table d sequence ("parent") mnemonic.
 C>
+C> @remark The interpretation of the return value iret depends upon the input value ient, as follows:
+C> ient=0 - iret = a count of the total number of child mnemonics within tabd(id,lun)
```
Comment (edwardhartnett) [inline src/uptdd.f]: Probably you want these two items to be bullet points. Try this:

```
C> @remark The interpretation of the return value iret depends upon the input value ient, as follows:
C> - ient=0 - iret = a count of the total number of child mnemonics within tabd(id,lun)
C> - ient>0 - iret = the bit-wise representation of the fxy value corresponding to the ient'th child mnemonic of tabd(id,lun)

```

Note the hyphens.

**Review by jbathegit — APPROVED** [review]

**Comment by jbathegit** [comment]
@edwardhartnett I'm ready to merge this, but merging is blocked because it says you still have some requested changes that need to be addressed.  Everything's been resolved now (I think), but if you still have some concerns could you please clarify, or else please remove the block on this merge?

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**